src/algebra/module/basic.lean

```diff
@@ -496,14 +496,14 @@ lemma function.injective.no_zero_smul_divisors {R M N : Type*} [has_zero R] [has
 ⟨λ c m h,
   or.imp_right (@hf _ _) $ h0.symm ▸ eq_zero_or_eq_zero_of_smul_eq_zero (by rw [←hs, h, h0])⟩
 
+instance no_zero_divisors.to_no_zero_smul_divisors [has_zero R] [has_mul R] [no_zero_divisors R] :
```

Comment:
This feels like it should be low priority as it's an instance derived from another instance, but clearly the linter doesn't care.